executor/executor_test.go

```diff
@@ -284,6 +284,84 @@ func (s *testSuite) TestSelectBackslashN(c *C) {
 	c.Check(fields[0].Column.Name.O, Equals, `\N`)
 }
 
+// Issue #3686
```

Comment:
1. better to remove this comment, since issue 3686 describes only string literal
2. add test for `'\\n'`